### PR TITLE
Hive crawler

### PR DESCRIPTION
**Pull request by usefulalgorithm**
### 🤔 Why?

To parse metadata information from Apache Hive.

### 🤓 What?

- Implemented basic hive crawler.
- Basic tests using simple hive tables.

### 🧪 Tested?

Tested end to end.

Hiveql tables:
截圖 2024-01-04 下午6 34 54

### ☑️ Checks

- [x] My PR contains actual code changes, and I have updated the version number in `pyproject.toml`.

### CODE REVIEW

**Comment by usefulalgorithm** [comment]
## Things I didn't do in this PR

- Parse materialized views and views
- Parse column stats and table stats

Finding out how to create them and put them in the test is not trivial. I think it's best to do those things in their own separate PRs.

- Config for kerberos
    - Don't know how to set this up, if this is needed then I'll do it

## Things I don't think are possible

- Get query logs
    - HiveQL simply does not let user get their query history. Looks like the only way to do that is to check inside the server host.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -33,6 +33,10 @@ jobs:
         with:
           poetry-version: ${{ env.POETRY_VERSION }}
 
+      - name: Setup dev libs
+        run: |
+          sudo apt-get install libsasl2-dev
```
Comment (mars-lan) [inline .github/workflows/ci.yml:38]: Do we need `libsasl2-dev` or just `libsasal2`?

File: Dockerfile [hunk 2]
```diff
@@ -3,7 +3,7 @@ FROM python:3.8-slim
 
 RUN apt-get clean
 RUN apt-get update
-RUN apt-get install -y git
+RUN apt-get install -y git build-essential libsasl2-dev
```
Comment (mars-lan) [inline Dockerfile:6]: Wny `build-essential`?

File: metaphor/hive/README.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# Hive Connector
+
+This connector extracts technical metadata from Apache Hive.
+
+## Setup
```
Comment (mars-lan) [inline metaphor/hive/README.md:5]: Let's move this to https://github.com/MetaphorData/data-systems?

File: Dockerfile (hunk 2 above)
Comment (usefulalgorithm) [reply Dockerfile:6]: when pip installing sasl it actually compiles it with gcc

File: metaphor/hive/README.md (hunk 3 above)
Comment (usefulalgorithm) [reply metaphor/hive/README.md:5]: will do that

File: .github/workflows/ci.yml (hunk 1 above)
Comment (usefulalgorithm) [reply .github/workflows/ci.yml:38]: we need dev, see https://github.com/MetaphorData/connectors/actions/runs/7412853556/job/20170520094?pr=743

File: metaphor/hive/README.md (hunk 3 above)
Comment (usefulalgorithm) [reply metaphor/hive/README.md:5]: see https://github.com/MetaphorData/data-systems/pull/4

File: metaphor/hive/README.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+To locally setup an Apache Hive service, run the following command:
+
+```shell
+docker-compose -f metaphor/hive/docker-compose.yml up -d
+```
+
+This sets up an Apache Hive service, along with several pre-populated table definitions.
```
Comment (alyiwang) [inline metaphor/hive/README.md:13]: Do we need to include this in the README? I assume the customer running the hive crawler should already have a hive instance.

File: metaphor/hive/README.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+## Config File
+
+Create a YAML config file based on the following template.
+
+### Required Configurations
+
+```yaml
+host: <host>
+port: <port>
+auth: <auth user for hiveserver>
```
Comment (alyiwang) [inline metaphor/hive/README.md:24]: maybe name it `auth_user` for clarity?

File: metaphor/hive/config.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+from metaphor.common.base_config import BaseConfig
+from metaphor.common.dataclass import ConnectorConfig
+
+
+@dataclass(config=ConnectorConfig)
+class HiveRunConfig(BaseConfig):
+    host: str
+    port: int
+    auth: Optional[str] = None
+    password: Optional[str] = None
```
Comment (alyiwang) [inline metaphor/hive/config.py:14]: Should those two be required?

File: metaphor/hive/README.md (hunk 4 above)
Comment (alyiwang) [reply metaphor/hive/README.md:13]: maybe move it to `testing` section

File: metaphor/hive/config.py (hunk 6 above)
Comment (usefulalgorithm) [reply metaphor/hive/config.py:14]: no, for a locally setup instance these aren't required

**Review by elic-eon — APPROVED** [review]
LGTM!

**Review by elic-eon — APPROVED** [review]

**Pull request merged**